prow/flagutil/plugins/plugins.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flagutil
+
+import (
+	"flag"
+	"fmt"
+
+	"k8s.io/test-infra/prow/flagutil"
+	"k8s.io/test-infra/prow/plugins"
+)
+
+const (
+	// DefaultConfigPath is the default for the plugin config path. It can be
+	// overridden by setting PluginConfigPathDefault.
+	DefaultConfigPath = "/etc/plugins/plugins.yaml"
+)
+
+type PluginOptions struct {
+	PluginConfigPath                         string
+	PluginConfigPathDefault                  *string
+	SupplementalPluginsConfigDirs            flagutil.Strings
+	SupplementalPluginsConfigsFileNameSuffix string
+	CheckUnknownPlugins                      bool
+}
+
+func (o *PluginOptions) AddFlags(fs *flag.FlagSet) {
+	defaultConfigPath := DefaultConfigPath
+	if o.PluginConfigPathDefault != nil {
+		defaultConfigPath = *o.PluginConfigPathDefault
```

Comment:
Nit: I think `AddFlags` should default to not specifying a `defaultConfigPath` so that `PluginConfigPathDefault` only needs to be changed to set a default. Since we want to encourage explicitly setting the plugin config path it seems better to not provide a default value unless the binary explicitly requests otherwise.

Reply:
Good point, it was mostly there because I was mirroring the status quo where we have it. I've inversed it so that everything that wants a default needs to explicitly configure the PluginOptions.